lib/matplotlib/backends/backend_qt5agg.py

```diff
@@ -56,6 +64,23 @@ def paintEvent(self, e):
         In Qt, all drawing should be done inside of here when a widget is
         shown onscreen.
         """
+
+        # As described in __init__ above, we need to be careful in cases with
+        # mixed resolution displays if dpi_ratio is changing between painting
+        # events.
+        if self._dpi_ratio_prev is None:
+            self._dpi_ratio_prev = self._dpi_ratio
+        elif self._dpi_ratio != self._dpi_ratio_prev:
+            # The easiest way to resize the canvas is to emit a resizeEvent
+            # since we implement all the logic for resizing the canvas for
+            # that event.
+            event = QtGui.QResizeEvent(self.size(), self.size())
+            # We use self.resizeEvent here instead of QApplication.postEvent
+            # since the latter doesn't guarantee that the event will be emitted
+            # straight away, and this causes visual delays in the changes.
+            self.resizeEvent(event)
```

Comment:
Apparently not - at least the paintEvent method does not get called again as a result.